### DISCUSSION TITLE
Sentences are punctuated imperfectly sometimes.

### DISCUSSION THREAD

**Discussion by otakutyrant** [openai/whisper#1271] [Q&A] [answered]
This is a slice of my generated srt file:

```
1
00:00:00,000 --> 00:00:14,640
Welcome to EnglishClass101.com's English in Three Minutes, the fastest, easiest, and

2
00:00:14,640 --> 00:00:16,920
most fun way to learn English.
```

I hope whisper can generate the whole sentence as below:

```
1
00:00:00,000 --> 00:00:16,920
Welcome to EnglishClass101.com's English in Three Minutes, the fastest, easiest, and most fun way to learn English.
```

I tried `whisper --model medium --word_timestamps True --max_line_width 80 a.mp4` but it did not work.

**Comment by ryanheise** [comment, marked as answer]
You can find a discussion here: #1243

**Comment by otakutyrant** [comment]
It seems that the officials recommend using third-party tools to handle sentences. I just wrote such a simple script to handle srt files only:

```
import pysrt
from pysrt import SubRipFile, SubRipItem

subs = pysrt.open('9.srt')
stack = []
for sub in subs:
    if stack and stack[-1].text[-1].isalpha():
        head = stack.pop()
        tail = sub
        new_text = head.text + ' ' + tail.text
        merged_sub = SubRipItem(len(stack) + 1, head.start, tail.end, new_text)
        stack.append(merged_sub)
    else:
        sub.index = len(stack) + 1
        stack.append(sub)
new_subs = SubRipFile(items=stack)
new_subs.save('new_9.srt', encoding='utf-8')
```

**Reply by dgoryeo** [reply]
Hi @otakutyrant , if I may ask, what does the script do? Meges line till a fullstop reached?

**Reply by otakutyrant** [reply]
@dgoryeo It iterates through all the lines. When reading a new line, the script checks whether the latest line ends with a letter. If so, it means that the latest line does not have a punctuation mark, and the script will concatenate the latest line and the new line.
